### PR TITLE
[linux-port] Enable VerifierTest for Unix

### PR DESCRIPTION
**Pull request by pow2clk**
Enable gtest support and convert path name to ascii for ifstream

### CODE REVIEW

File: tools/clang/unittests/HLSL/TestMain.cpp [hunk 1]
```diff
@@ -142,6 +140,8 @@ int main(int argc, char **argv) {
 #endif
 #endif
 
+  moduleSetup();
   int rv = RUN_ALL_TESTS();
+  moduleTeardown();
```
Comment (pow2clk) [inline tools/clang/unittests/HLSL/TestMain.cpp]: The fact that this is done here is largely historical. Previously, these functions were implemented in VerifierTest.cpp.

**Comment by pow2clk** [comment]
The brevity of this change severely misrepresents how hard this particular test was to get working. It's true of many of these, but this one in particular. The tricky changes just went before so these are mostly just flipping a switch after completing all the wiring.

**Comment by ehsannas** [comment]
I applaud your efforts to get this working. I know how tricky these can be... 👏

**Comment by ehsannas** [comment]
More tests running on Unix:
```
[----------] 41 tests from VerifierTest
[----------] 41 tests from VerifierTest (548 ms total)
```

**Pull request merged**